src/ice_shelf/MOM_ice_shelf_dynamics.F90

```diff
@@ -2481,22 +2566,89 @@ subroutine calc_shelf_visc(CS, ISS, G, US, u_shlf, v_shlf)
   gisc = G%domain%nihalo+1 ; gjsc = G%domain%njhalo+1
   giec = G%domain%niglobal+gisc ; gjec = G%domain%njglobal+gjsc
   is = iscq - 1; js = jscq - 1
+    i_off = G%idg_offset ; j_off = G%jdg_offset
 
   n_g = CS%n_glen; eps_min = CS%eps_glen_min
 
-  Visc_coef = US%kg_m2s_to_RZ_T*US%m_to_L*US%Z_to_L*(CS%A_glen_isothermal)**(1./CS%n_glen)
-
-  do j=jsd+1,jed-1
-    do i=isd+1,ied-1
-
-      if (ISS%hmask(i,j) == 1) then
-        ux = ((u_shlf(I,J) + u_shlf(I,J-1)) - (u_shlf(I-1,J) + u_shlf(I-1,J-1))) / (2*G%dxT(i,j))
-        vx = ((v_shlf(I,J) + v_shlf(I,J-1)) - (v_shlf(I-1,J) + v_shlf(I-1,J-1))) / (2*G%dxT(i,j))
-        uy = ((u_shlf(I,J) + u_shlf(I-1,J)) - (u_shlf(I,J-1) + u_shlf(I-1,J-1))) / (2*G%dyT(i,j))
-        vy = ((v_shlf(I,J) + v_shlf(I-1,J)) - (v_shlf(I,J-1) + v_shlf(I-1,J-1))) / (2*G%dyT(i,j))
+!  CS%ice_visc(:,:) = 0.0
+!  ux(:,:) = 0.0; uy(:,:) = 0.0; vx(:,:) =0.0; vy(:,:) =0.0
+!  eII(:,:) = (US%s_to_T**2 * (eps_min**2))
+  Visc_coef = US%kg_m2s_to_RZ_T*US%m_to_L*US%Z_to_L*(CS%A_glen_isothermal)**(-1./CS%n_glen) !OVS '-' in the exponent
+!    call pass_vector(u_shlf, v_shlf, G%domain, TO_ALL, BGRID_NE)
+    do j=jsc-0*1,jec+0*1
+    do i=isc-0*1,iec+0*1
+!  do j=jsd+1,jed-1     !OVS 02/01/21
+!    do i=isd+1,ied-1   !OVS 02/01/21
+
+      if ((ISS%hmask(i,j) == 1) .OR. (ISS%hmask(i,j) == 3)) then
+!        ux(i,j) = ((u_shlf(I,J) + u_shlf(I,J-1)) - (u_shlf(I-1,J) + u_shlf(I-1,J-1))) / (2*G%dxT(i,j))
+!        vx(i,j) = ((v_shlf(I,J) + v_shlf(I,J-1)) - (v_shlf(I-1,J) + v_shlf(I-1,J-1))) / (2*G%dxT(i,j))
+!        uy(i,j) = ((u_shlf(I,J) + u_shlf(I-1,J)) - (u_shlf(I,J-1) + u_shlf(I-1,J-1))) / (2*G%dyT(i,j))
+!        vy(i,j) = ((v_shlf(I,J) + v_shlf(I-1,J)) - (v_shlf(I,J-1) + v_shlf(I-1,J-1))) / (2*G%dyT(i,j))
+!      endif
+!      enddo
+!      enddo
+!    call pass_vector(ux, uy, G%domain, TO_ALL, BGRID_NE)
+!    call pass_vector(vx, vy, G%domain, TO_ALL, BGRID_NE)
+        ux = ((u_shlf(I,J) + u_shlf(I,J-1) + u_shlf(I,J+1)) - &
```

Comment:
The sums of three terms here without parentheses will change answers if the problem is rotated, and the rotational invariance test will fail.  I suggest adding parentheses to group the last two terms in each sum, to that the code will reproduce answers under a rotation of 180-degrees.

  I would note that because this code has not been exercise